pandas/core/indexes/multi.py

```diff
@@ -2676,10 +2676,17 @@ def _partial_tup_index(self, tup, side="left"):
                 return start + section.searchsorted(loc, side=side)
 
             idx = self._get_loc_single_level_index(lev, lab)
-            if k < n - 1:
+            if isinstance(idx, slice) and k < n - 1:
+                # Get start and end value from slice, necessary when a non-integer
+                # interval is given as input
+                start = idx.start
+                end = idx.stop
+            elif k < n - 1:
                 end = start + section.searchsorted(idx, side="right")
                 start = start + section.searchsorted(idx, side="left")
             else:
+                if isinstance(idx, slice):
```

Comment:
something is wrong here, L2684 doesn't actually return anything, nor does the clause you added because L2690 is not indented correctly.

Reply:
I am not sure if I understand you correctly, but the intendation is on purpose. 

We are within a loop and hit the else clause in the last pass through. That is the only time we want to return anything. Additionally the return should not be within ``if isinstance(idx, slice)``. Just want to set the input for ``section.searchsorted`` to the start value of the slice if we have one.